### ISSUE TITLE
Javadoc Trees should be synchronized with the current checkstyle version

### ISSUE THREAD

**Issue by ps-sp** [checkstyle/checkstyle#4640]
Taken from [4408 PR comment](https://github.com/checkstyle/checkstyle/pull/4550#discussion_r125208278). The javadoc trees should be updated to reflect all the new ANTLR changes being incorporated into checkstyle. This issue should be approved or be worked upon once all the major ANTLR updates are done with.

**Comment by rnveach** [comment]
@ps-sp update issue with more detail on which tree are you talking about. The more details in issue, the more searching someone in the future doesn't need to do.

It would be great if we could automate the synchronization somehow so we don't forget this in the future.

**Comment by ps-sp** [comment]
@rnveach 

>It would be great if we could automate the synchronization 

Correct me if I am wrong, this sounds like one of the project ideas which was about generating xdocs from java documentation. I can try to work on that idea after GSoC.

> update issue with more detail on which tree

Actually I didn't realize that you would be approving this issue right away and had created this just as a placeholder until ANTLR updates are done with. I would've updates the description later which would have helped me to point more accurately to the places the trees needed to be updated. Anyway, now that it is approved, I have updated the issue description a little bit.

**Comment by rnveach** [comment]
>  this sounds like one of the project ideas which was about generating xdocs from java documentation though of course not exactly the same

Yes, sorta. If it is too much trouble, we can skip for now, but it will come up again.
It can be very rough process if needs be to get us started.

> I didn't realize that you would be approving this issue right away

I will approve things if I believe they should be done no matter what. I assume @romani will agree here too that this file should be in synch.
Approval doesn't mean it needs to be done now. GSoC tag will prevent anyone else from working on it.
Issue can be updated more later if more things come to light. It should contain all information known now so we don't forget.

**Comment by ps-sp** [comment]
@rnveach acknowledged.

**Comment by romani** [comment]
I removed all labels as this is hard to automate process and we do not change grammar that often, special project that @ps-sp  has is not countable.

Changes is JavadocTokenTypes.java  is not avoidable and it is good to make them during PR to clearly define the change.
Yes, it will be often a point to forget to update, but validation of sync is not trivial as we do not have standard format of such javadoc comments.  One day, one day .....

**Comment by romani** [comment]
Not approved as overcomplicated to make , we might come back to it if there will be demand for this
